pkg/backend/httpstate/token_source.go

```diff
@@ -89,9 +89,10 @@ func (ts *tokenSource) handleRequests(
 	defer renewTicker.Stop()
 
 	state := struct {
-		token   string    // most recently renewed token
-		error   error     // non-nil indicates a terminal error state
-		expires time.Time // assumed expiry of the token
+		token    string    // most recently renewed token
+		error    error     // non-nil indicates a terminal error state
+		expires  time.Time // assumed expiry of the token
+		errorLog string    // log that's displayed to the user in case of error renewing the token for debugging purposes
```

Comment:
instad of a string does it make sense to store an `error` and use %w to combine them so that we can use errors.Is elsewhere etc?  I don't like stringly typing